hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,329 @@
+            Shows progressbar while iterating over different slices of the signal (passes the parameter to map method).
+        log_parameters : boolean
+            Logs reconstruction parameters
+
+        Returns
+        -------
+        wave : :class:`~hyperspy.signals.WaveImage
+            Reconstructed electron wave. By default object wave is devided by reference wave
+
+        Notes
+        -----
+
+        """
+
+        # Parsing reference:
+        if not isinstance(reference, HologramImage):
+            if isinstance(reference, Signal2D):
+                _logger.warning('The reference image signal type is not HologramImage. It will '
+                                'be converted to HologramImage automatically.')
+                reference.set_signal_type('hologram')
+            elif reference is not None:
+                reference = HologramImage(reference)
+
+        # Testing match of navigation axes of reference and self (exception: reference nav_dim=1):
+        if (reference and not reference.axes_manager.navigation_shape == self.axes_manager.navigation_shape
+            and reference.axes_manager.navigation_size):
+
+            raise ValueError('The navigation dimensions of object and reference holograms do not match')
+
+        # Parsing sideband position:
+        if sb_position is None:
+            warnings.warn('Sideband position is not specified. The sideband will be found automatically which may '
+                          'cause wrong results.')
+            if reference is None:
+                sb_position = self.find_sideband_position(sb=sb)
+            else:
+                sb_position = reference.find_sideband_position(sb=sb)
+
+        else:
+            if not isinstance(sb_position, Signal1D):
+                sb_position = Signal1D(sb_position)
+
+        if sb_position.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_position.axes_manager.navigation_size:
+                warnings.warn('Sideband position dimensions do not match neither reference nor hologram dimensions.'
+                              'The reconstruction will be performed with the first values')
+                sb_position_temp = sb_position.inav[0].data
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_position_temp = sb_position.data
+        else:
+            sb_position_temp = sb_position.deepcopy()
+        #
+
+        # Parsing sideband size
+        if sb_size is None:  # Default value is 1/2 distance between sideband and central band
+            if reference is None:
+                sb_size = self.find_sideband_size(sb_position)
+            else:
+                sb_size = reference.find_sideband_size(sb_position)
+        else:
+            if not isinstance(sb_size, BaseSignal):
+                if isinstance(sb_size, np.ndarray) and sb_size.size > 1:  # transpose if np.array of multiple instances
```

Comment:
Here I'd rather prefer keeping possibility to input arrays, therefore added corresponding changes in docstring. Transposition seem to be required in any case as `sb_size` signal dimension should be 0, and BaseSignal will always put all the dimensions of array into signal_dimension. The case when user messes up the navigation dimensions in the input array I'd leave as user's risk.